sql/mysqld.h

```diff
@@ -174,15 +174,34 @@ extern char *opt_backup_history_logname, *opt_backup_progress_logname,
             *opt_backup_settings_name;
 extern const char *log_output_str;
 extern const char *log_backup_output_str;
-extern char *temporal_current_timestamp;
-extern my_bool vers_force;
+
+enum vers_range_type_t
+{
+  FOR_SYSTEM_TIME_UNSPECIFIED = 0,
+  FOR_SYSTEM_TIME_AS_OF,
+  FOR_SYSTEM_TIME_FROM_TO,
+  FOR_SYSTEM_TIME_BETWEEN,
+  FOR_SYSTEM_TIME_ALL,
+  FOR_SYSTEM_TIME_BEFORE
+};
+
+struct st_vers_current_time
+{
+  char *str_value; // must be first
+  vers_range_type_t type;
+  MYSQL_TIME ltime;
+  st_vers_current_time() :
```

Comment:
A lot of warnings because of this ctor:
```
/home/kevg/work/mariadb_10.2/sql/sys_vars.cc:5556:8: warning: offset of on non-POD type 'SV' (aka 'system_variables') [-Winvalid-offsetof]
       SESSION_VAR(session_track_state_change),
       ^           ~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/kevg/work/mariadb_10.2/sql/sys_vars.ic:43:42: note: expanded from macro 'SESSION_VAR'
#define SESSION_VAR(X) sys_var::SESSION, offsetof(SV, X), sizeof(((SV *)0)->X)
                                         ^            ~
/usr/lib/llvm-3.8/bin/../lib/clang/3.8.1/include/stddef.h:120:24: note: expanded from macro 'offsetof'
#define offsetof(t, d) __builtin_offsetof(t, d)
                       ^                     ~
```

It can be delete thanks to [this code](https://github.com/tempesta-tech/mariadb_10.2/blob/2a767b490034d3288cb70daed9607336369beea5/sql/sql_class.cc#L808):
```c++
  connection_name.str= 0;
  connection_name.length= 0;

  bzero(&variables, sizeof(variables));
  file_id = 0;
  query_id= 0;
```